### PR TITLE
ML: each Jobs test creates a datasource to depend on

### PR DESCRIPTION
**Pull request by gerboland**
Create a datasource and use it for each ML job. Avoids depending on magic datasource id/uids.

### CODE REVIEW

**Comment by julienduchesne** [comment]
> The datasource ID of 10 is invalid, causing the ML job to fail with an error that impacts our SLO.
> 
> This changes it to 11 which is valid.

Can we provision the datasource in the test rather than using a magic number?

**Comment by gerboland** [comment]
> Can we provision the datasource in the test rather than using a magic number?

Not a bad idea, I'll look into that. Would you know if there's prior art for that?

**Comment by julienduchesne** [comment]
> > Can we provision the datasource in the test rather than using a magic number?
> 
> Not a bad idea, I'll look into that. Would you know if there's prior art for that?

You can create a datasource like this and reference it: https://github.com/grafana/terraform-provider-grafana/blob/main/internal/resources/grafana/resource_data_source_permission_test.go#L69-L82

**Comment by gerboland** [comment]
Thanks, I'll get that going

File: internal/resources/machinelearning/resource_job_test.go [hunk 1]
```diff
@@ -18,53 +18,61 @@ func TestAccResourceJob(t *testing.T) {
...
+			testAccMLJobCheckDestroy(&job),
+			testAccDatasourceCheckDestroy(),
+		),
 		Steps: []resource.TestStep{
 			{
-				Config: testutils.TestAccExampleWithReplace(t, "resources/grafana_machine_learning_job/job.tf", map[string]string{
+				// Note for the reader: these tests construct a datasource & a job, where the Job's datasource id is set by terraform.
+				Config: machinelearningDatasource + testutils.TestAccExampleWithReplace(t, "resources/grafana_machine_learning_job/job.tf", map[string]string{
 					"Test Job": randomName,
+					"= 10":     "= split(\":\", grafana_data_source.foo.id)[1]", // as it of form ^(0|1):[a-zA-Z0-9-_]+$, we only use later part
```
Comment (julienduchesne) [inline internal/resources/machinelearning/resource_job_test.go:35]: It'd be good if ML supported uids as IDs are being deprecated but that's for another Pr

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**